src/main/java/com/extremelyd1/game/progress/ProgressController.java

```diff
@@ -0,0 +1,45 @@
+package com.extremelyd1.game.progress;
+
+import org.bukkit.configuration.file.FileConfiguration;
+
+import java.util.List;
+
+public class ProgressController {
+
+    /**
+     * Rules for lines notifications
+     */
+    private List<String> linesRules;
+
+    /**
+     * Rules for full card count notifications
+     */
+    private List<String> fullCardCountRules;
+
+    /**
+     * Rules for lockout gamemode
+     */
+    private List<String> lockoutRules;
+
+    public ProgressController(FileConfiguration config) {
+        linesRules = config.getStringList("notify-progress.lines");
+        fullCardCountRules = config.getStringList("notify-progress.full-card");
+        lockoutRules = config.getStringList("notify-progress.lockout");
+    }
+
+    private Boolean notifyEachLine() {
+        return linesRules.contains("each");
+    }
+
+    public Boolean shouldNotifyProgressLines(int linesComplete) {
+        return linesRules.contains(String.valueOf(linesComplete)) || notifyEachLine();
+    }
+
+    public Boolean shouldNotifyCountAmount(int itemsCollected) {
+        return fullCardCountRules.contains(String.valueOf(itemsCollected));
+    }
+
+    public Boolean shouldNotifyLockoutAmount(int itemsCollected) {

```

Comment:
This method is unused, which probably shouldn't be the case.

Reply:
This was a mistake elsewhere which has been resolved